presentation/src/main/java/com/sopt/presentation/ui/util/AndroidExtensions.kt

```diff
@@ -0,0 +1,12 @@
+package com.sopt.presentation.ui.util
+
+import android.os.Build
+import android.os.Bundle
+
+internal inline fun<reified T> Bundle.getParcelableCompat(key: String): T? {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        getParcelable(key, T::class.java)
+    } else {
+        getParcelable(key)
+    }
+}
```

Comment:
최신 버전들을 사용하고 계신다면 (androidx.core 라이브러리였던거같은데)
아래와 같은 방식으로 사용 가능합니다. 지금도 사용 가능해보여요!

```suggestion
inline fun <reified T : Parcelable> Bundle.parcelable(key: String): T? {
    return BundleCompat.getParcelable(this, key, T::class.java)
}

```

Reply:
참고로 경험하시지 않았으면 좋겠지만 java 클래스에서 해당 값을 쓸 경우에는 아래와 같은 이유로 코드를 다르게 짜야할 수도 있습니다!

```kotlin
fun <T : Parcelable> parcelable(bundle: Bundle, key: String, clazz: Class<T>): T? {
    return BundleCompat.getParcelable(bundle, key, clazz)
}
```

> Java 의 경우, inline 및 reified 키워드를 제공하지 않기 때문에
> 제너릭 타입 T 에 대한 정보는 런타임 단계에서 Type Erase 되어버림.
> 따라서, 파라미터로 Class<T> 타입을 전달하여 사용해야 한다.

Reply:
굉장히 좋은 게 있었군요...! 감사합니다.

+자바에 대해서는 한 번도 신경안써봤는데, 이런 경우도 있군요..! 기억해두겠습니다!